### PR TITLE
Add 18comic

### PR DESCRIPTION
**Pull request by MikeZeDev**
https://github.com/manga-download/hakuneko/issues/2708

### CODE REVIEW

File: web/src/engine/websites/EighteenComic.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    public constructor() {
+        super('18comic', '18comic', 'https://18comic.org', Tags.Media.Manga, Tags.Language.Chinese, Tags.Source.Aggregator, Tags.Rating.Pornographic);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const dom = await FetchHTML(new Request(new URL(manga.Identifier, this.URI)));
```
Comment (ronny1982) [inline web/src/engine/websites/EighteenComic.ts:52]: Consider using `FetchCSS` instead

Comment (ronny1982) [reply web/src/engine/websites/EighteenComic.ts:52]: Or even `Common.FetchChaptersSinglePageCSS` with an appropriate `AnchorInfoExtractor`.

File: web/src/engine/websites/EighteenComic.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        const pages = await FetchWindowScript<PageSeed[]>(new Request(new URL(chapter.Identifier, this.URI)), pageScript, 1500);
+        return pages.filter(page => page.url)
+            .map(page => new Page<Seed>(this, chapter, new URL(page.url), { seed: page.seed }));
+    }
+
+    public override async FetchImage(page: Page<Seed>, priority: Priority, signal: AbortSignal): Promise<Blob> {
+        const blob = await Common.FetchImageAjax.call(this, page, priority, signal);
+        const seed = page.Parameters.seed;
+        return seed < 0 ? blob : DeScramble(blob, async (image, ctx) => {
+            const l = image.height % seed ;
```
Comment (ronny1982) [inline web/src/engine/websites/EighteenComic.ts:71]: This will fail when `seed = 0`, is this by intention?

File: web/src/engine/websites/EighteenComic.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    return {
+        id: element.querySelector<HTMLAnchorElement>('a').pathname,
+        title: element.querySelector<HTMLImageElement>('img').alt.trim()
+    };
+}
+
+const pageScript = `
+    new Promise ( resolve => {
+        const imagenodes = [...document.querySelectorAll('div.row.thumb-overlay-albums img')];
+        const scrambled = aid > scramble_id;
```
Comment (ronny1982) [inline web/src/engine/websites/EighteenComic.ts:18]: May remove intermediate variable if only used once

File: web/src/engine/websites/EighteenComic.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        resolve ( imagenodes.map(image => {
+            const imageNumber = image.parentNode.id.split('.')[0];
+            const seed = scrambled ? get_num(window.btoa(aid), window.btoa(imageNumber)) : -1;
```
Comment (ronny1982) [inline web/src/engine/websites/EighteenComic.ts:21]: May return `0` instead of `-1` to avoid confusion that `0` could be a valid seed

**Review by ronny1982 — APPROVED** [review]
